### PR TITLE
CPT-80 ui: use new header and footer layout

### PR DESCRIPTION
**Pull request by 634750802**
blocked by #36, #38

break change:
-  Add **required** `hasMargin` prop to common `Footer` component controlling margin instead of using default margin

### CODE REVIEW

File: packages/pingcap-ui/src/components/footer/Footer.component.js
Comment (hustKiwi) [inline packages/pingcap-ui/src/components/footer/Footer.component.js:22]: It's not the suggested pattern of using styled components because the container change will cause the component rerendering.

Typically, we will adapt stylings by passing a property to the styled component, like https://styled-components.com/docs/basics#adapting-based-on-props

```jsx
<Styled.Container hasMargin={hasMargin}>
  // ...
</Styled.Container
```
Then, in the `footer.styled.js`:
```javascript

export const Container = styled.div`
  background: ${colors.M2};
  ${props => props.hasMargin && css`
    margin-top: 6rem;
  `;
`;
```

File: src/layouts/core/Core.component.js
Comment (hustKiwi) [inline src/layouts/core/Core.component.js:26]: Use es6 shortband:

```javascript
  const headerProps = {
    logo,
    title,
    onNavClick,
    hasMargin,
    navItems: headerNavItems,
    currentNav: utils.header.getCurrentNav(headerNavItems, router.pathname),
    onTitleClick: () => router.push('/community'),
  };
```

**Review by hustKiwi — CHANGES_REQUESTED** [review]

File: packages/pingcap-ui/src/components/footer/Footer.component.js
Comment (634750802) [reply packages/pingcap-ui/src/components/footer/Footer.component.js:22]: Thanks, I'm not familiar with styled component yet.

Comment (hustKiwi) [reply packages/pingcap-ui/src/components/footer/Footer.component.js]: No worries, that's the value of code review :) Practice makes perfect!

**Comment by 634750802** [comment]
Please don't merge this PR before #36 was merged.

**Review by hustKiwi — APPROVED** [review]

**Comment by 634750802** [comment]
test

**Comment by 634750802** [comment]
@hustKiwi I have implemented a simple solution for i18n on header and footer. Please review it. 

New `header.data.js` content was changed to:

```js
import * as dataEn from './footer.data-en';
import * as dataCn from './footer.data-cn';

export default {
  en: dataEn,
  cn: dataCn,
};

export * from './footer.data-en';
```

The last statement is aimed to be compatible with previous usage. And the `getData` was changed to:

```js

...

import footerData, * as defaultFooterData from './components/footer/footer.data';
import headerData, * as defaultHeaderData from './components/header/header.data';

...

export const getData = (domain, path, locale) => {
  const urlPrefixRegexp = nav.buildUrlPrefixPattern(domain, path);

  const { navItems: footerNavItems, ...restFooterData } = footerData[locale] || defaultFooterData;
  const { navItems: headerNavItems, ...restHeaderData } = headerData[locale] || defaultHeaderData;

  return {
    footer: {
      navItems: nav.replaceNavLinks(footerNavItems, urlPrefixRegexp),
      ...restFooterData,
    },
    header: {
      navItems: nav.replaceNavLinks(headerNavItems, urlPrefixRegexp),
      ...restHeaderData,
    },
  };
};

...

```

File: src/layouts/core/Core.component.js
Comment (hustKiwi) [inline src/layouts/core/Core.component.js:50]: Better to strickly check all available options:

```javascript
locale: PropTypes.oneOf(['cn', 'en'])
```

File: packages/ui/src/components/footer/footer.data.js
Comment (hustKiwi) [inline packages/ui/src/components/footer/footer.data.js:2]: Good catch of CI failing :)

File: packages/ui/src/components/header/header.data.js
Comment (hustKiwi) [inline packages/ui/src/components/header/header.data.js:2]: I am not quite sure whether is a good practice to mix-use camelCase & kebab-case for filenames. But as a temporary solution before introducing a solid i18n approach, it should be all good. No change is required.

File: packages/ui/src/components/header/header.data.js
Comment (hustKiwi) [inline packages/ui/src/components/header/header.data.js:9]: Is it required? If we have other items which are unrelated to the i18n stuff, I guess we should store them in this footer.data.js, right?

Comment (634750802) [reply packages/ui/src/components/header/header.data.js:9]: This is aimed to be compatible with previous usage. For example, in the storybook, the old usage was not changed.

**Comment by 634750802** [comment]
We need extra required peer dependencies 'ramda', 'polished'

**Comment by hustKiwi** [comment]
@634750802 Yes, `ramda` and `polished` could be added as peer dependencies. But you may do that in a separate PR for CPT-82. We need to scope down the changes for each PR. I think for this PR, it's good to go.

**Pull request merged**
